### DISCUSSION TITLE
How to use IDFVersionUpdater correctly

### DISCUSSION THREAD

**Discussion by j-roh** [NREL/EnergyPlus#9374] [Q&A] [answered]
How can I convert the EnergyPlus version of idf file?

I was trying to convert [DOE reference building idf file](https://www.energy.gov/eere/buildings/downloads/reference-buildings-building-type-medium-office) to V9.3 by using the IDFVersionUpdater.

The below is what I tried with "Create Intermediate Version Files" checked, and all of them failed.

(1) convert V7.2 to V9.3 using IDFVersionUpdater found in EnergyPlusV9-3-0_PreProcess_IDFVersionUpdater
(2) convert V7.2 to V9.3 using IDFVersionUpdater found in Utilies_EP-Launch
(3) convert V5.0 to V9.3 using IDFVersionUpdater after [installing additional files](https://energyplushelp.freshdesk.com/support/solutions/articles/67000682946-windows-programs-for-converting-older-pre-v7-2-0-idf-files-to-current-or-intermediate-versions-) in the same folder
I also tried using V9.3 idf file created after trying
(4) convert V7.2 to V9.4 using IDFVersionUpdater found in EnergyPlusV9-4-0_PreProcess_IDFVersionUpdater
(5) convert V7.2 to V9.4 using IDFVersionUpdater found in Utilies_EP-Launch
(6) convert V5.0 to V9.4 using IDFVersionUpdater after installing additional files in the same link in (2)

Even though 
"Completed transition of: ~RefBldgMediumOfficePost1980_v1.3_5.0_1A_USA_FL_MIAMI.idf From 5.0.0/7.2.0 to 9.3.0/9.4.0" [pops up](https://imgur.com/a/sfA9lIj),
and [whole intermediate version files are created but it never works as shown in the screenshots.](https://imgur.com/a/lZOKORw) 

I would really appreciate any help to solve this version update problem.

Thank you in advance.

**Comment by mjwitte** [comment, marked as answer]
@j-roh I cannot reproduce this locally (on Windows 10). IDFVersionUpdater successfully updates the file to v9.3.0 without errors. Your file explorer screenshot shows the exact same date/time on all versions of the file, indicating the updater did nothing except copy the files. What version of Windows are you using? Make sure that you have full read/write/modify/execute permissions to the working folder. Also, make sure the full path to the files is not longer than 128 characters (I think that's the limit in some versions of windows). Does the .audit file give any clues?

I tested this with IDFVersionUpdater from v9.3.0 (with and without saving intermediate versions), v9.6.0, and v22.1.0. They all worked fine.
[Transition to 9.3 save intermediates.zip](https://github.com/NREL/EnergyPlus/files/8483006/Transition.to.9.3.save.intermediates.zip)
 
In the future, please use one of the [EnergyPlus User Support options](https://energyplus.net/support-training) for questions like this. Thanks.

**Reply by j-roh** [reply]
Thank you for your reply and introduction to the support options that I can utilize from now on.

I am currently using Windows Version 10 (21H2),

I found out that my initial location of the idf file has over 128 characters (about 140) so I've changed the location under EnergyPlusV9-3-0_ExampleFiles and the IDFVersionUpdater did work.

I really do appreciate your help.

**Reply by j-roh** [reply]
Oh, and the audit file showed the code "File not found=C:~(the initial path that was long)".

Thanks again for your help.
